### PR TITLE
chore: add name column in dashboards table for v2 dashboards

### PR DESCRIPTION
**Pull request by namanv3**
## Pull Request

---

### 📄 Summary
Reviews for https://github.com/SigNoz/signoz/pull/11125 will take some time, and migration numbers keep on increasing, so planning to merge just the migration part first.
---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [ ] 🐛 Bug fix
- [ ] ♻️ Refactor
- [x] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

- Tests added/updated:
- Manual verification: Done, v1 dashboards created before and after this migration. No issues seen.
- Edge cases covered:

---

- Blast radius: Dashboards table
- Potential regressions: None, new column not filled by anything yet
- Rollback plan: Delete column from the table

---

### CODE REVIEW

File: pkg/sqlmigration/089_add_dashboard_name.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	// Only v2 dashboards populate this column. Existing v1 rows are left with
+	// the zero value (empty string) so v1 create/update paths can keep
+	// inserting without a name.
+	//
+	// TODO: once v1 dashboards are migrated to v2 and every row has a real
+	// name, a follow-up migration should add a unique index on
+	// (org_id, name) to enforce per-org name uniqueness.
+	sqls := migration.sqlschema.Operator().AddColumn(table, uniqueConstraints, nameColumn, nil)
```
Comment (therealpandey) [inline pkg/sqlmigration/089_add_dashboard_name.go:64]: @namanv3 is this migration tested?

When nullable is false, is the operator running `UPDATE dashboard SET name = ''` automatically?

**Review by therealpandey — CHANGES_REQUESTED** [review]

File: pkg/sqlmigration/089_add_dashboard_name.go (hunk 1 above)
Comment (namanv3) [reply pkg/sqlmigration/089_add_dashboard_name.go:64]: yes, tested with sqlite. let me just run the integration tests here so that Postgres is also confirmed

**Review by therealpandey — APPROVED** [review]

**Review by srikanthccv — APPROVED** [review]
